### PR TITLE
new intake special movements

### PR DESCRIPTION
**Pull request by LandonBayer**
this isn't necessarily ready to be pushed in yet, but I'd like y'all to check over it anyway

### CODE REVIEW

File: subsystems/intakesubsystem.py [hunk 1]
```diff
@@ -161,7 +157,7 @@ def periodic(self) -> None:
                 * constants.kPivotGearRatio,
             )
             self.intakeMotor.set(
-                NEOBrushless.ControlMode.Velocity, constants.kIntakeSpeed * -1
+                NEOBrushless.ControlMode.Velocity, constants.kIntakePercentageVoltage * -1
```
Comment (lmaxwell24) [inline subsystems/intakesubsystem.py]: we should have a different ejecting voltage than intaking

File: subsystems/intakesubsystem.py [hunk 2]
```diff
@@ -90,25 +94,17 @@ def periodic(self) -> None:
...
-                    NEOBrushless.ControlMode.Position, self.heldPosition
-                )
-            else:
-                self.intakeMotor.set(
-                    NEOBrushless.ControlMode.Velocity, constants.kIntakeSpeed
-                )
-            self.intakeMotor.set(NEOBrushless.ControlMode.Position, self.heldPosition)
+
+            if self.putInPlace:
+                self.intakeMotor.set(NEOBrushless.ControlMode.Position, self.heldPosition)
```
Comment (lmaxwell24) [inline subsystems/intakesubsystem.py]: shouldn't this hold position logic exist in other held areas?

**Review by lmaxwell24 — CHANGES_REQUESTED** [review]

File: subsystems/intakesubsystem.py (hunk 2 above)
Comment (LandonBayer) [reply subsystems/intakesubsystem.py]: Yeah, this is far from done but wanted to see if this seems about right for the general idea

File: subsystems/intakesubsystem.py [hunk 3]
```diff
@@ -81,30 +85,33 @@ def periodic(self) -> None:
...
                 self.intakeMotor.set(
                     NEOBrushless.ControlMode.Position, self.heldPosition
                 )
-            elif (
-                not frontLimitState and backLimitState
-            ):  # Note: not only aplies to first part (this is for front off back on)
+            elif frontLimitState and not backLimitState:
                 self.intakeMotor.set(
-                    NEOBrushless.ControlMode.Position, self.heldPosition
+                    NEOBrushless.ControlMode.Percent, constants.kIntakePercentageVoltage
```
Comment (lmaxwell24) [inline subsystems/intakesubsystem.py]: should probably be slower here for fine control, like 1/2 speed

**Review by lmaxwell24 — CHANGES_REQUESTED** [review]

File: constants.py [hunk 4]
```diff
@@ -765,13 +765,18 @@
 # relative to horizontal
 kHandoffAngle = Rotation2d.fromDegrees(-7.761653)
 kFloorPositionAngle = Rotation2d.fromDegrees(217.28378)
-kStagingPositionAngle = Rotation2d.fromDegrees(102.299316)
-kAmpScoringPositionAngle = kStagingPositionAngle + Rotation2d.fromDegrees(5)
+kStagingPositionAngle = Rotation2d.fromDegrees(
+    50
+)  # Safe place roughly vertical before going to amp or trap
+kAmpScoringPositionAngle = Rotation2d.fromDegrees(69.939031)
+kStagingPositionangle = kAmpScoringPositionAngle
```
Comment (icai0) [inline constants.py]: you set this constant to 50 degrees and then redefine it

File: subsystems/intakesubsystem.py [hunk 5]
```diff
@@ -68,6 +69,9 @@ def periodic(self) -> None:
         frontLimitState = self.intakeMotor.getLimitSwitch(self.frontSensor)
         backLimitState = self.intakeMotor.getLimitSwitch(self.backSensor)
         self.hasPosition = backLimitState or frontLimitState
+        if not self.hasPosition:
+            self.putInPlace = False
+
         if self.state == self.IntakeState.Intaking:
```
Comment (icai0) [inline subsystems/intakesubsystem.py:75]: there's nothing in this state that tells the intake motor to run inwards, and there should also be logic for hasposition

**Review by icai0 — CHANGES_REQUESTED** [review]

File: subsystems/intakesubsystem.py (hunk 5 above)
Comment (LandonBayer) [reply subsystems/intakesubsystem.py:75]: Intake running is done by voltage control, so it's assumed we have the inversion true/false the way that it is irl, also has position does have logic (it's true when either sensor is on, the line right above)

**Review by lmaxwell24 — DISMISSED** [review]

**Review by lmaxwell24 — APPROVED** [review]

**Review by icai0 — APPROVED** [review]

**Pull request merged**